extensions/wikia/HTTPSSupport/HTTPSSupportHooks.class.php

```diff
@@ -70,8 +70,9 @@ public static function parserUpgradeSpecialFilePathURLs( string &$url ) {
 	 * @return boolean
 	 */
 	public static function onLinkerMakeExternalLink( string &$url, string &$text, bool &$link, array &$attribs ): bool {
-		if ( wfHttpsAllowedForURL( $url ) ) {
-			$url = wfProtocolUrlToRelative( $url );
+		global $wgWikiaBaseDomainRegex;
+		if ( preg_match( '/'. $wgWikiaBaseDomainRegex .'/', $url ) ) {
```

Comment:
This doesn't check the number of parts of the domain, so will this potentially convert multiple subdomain URLs to wikis (e.g. old links to `ja.starwars.wikia.com` for example) to HTTPS? If so, we should be excluding those as they do not support HTTPS so the links would be broken when clicked.

Reply:
Good point!
So overall we should:
* ignore third party urls here
* normalize urls (to support preview/vertify etc)
* for wgFandomBaseDomain and wgWikiaOrgBaseDomain use https
* for wgWikiaBaseDomain check the number of subdomain before using https

This seems like a nice method to have some clean unit tests :)